src/lib/work_partitioner/pending_zkapp_command.ml

```diff
@@ -1,15 +1,19 @@
-(* NOTE: Assumption: order of merging segment proofs is irrelvant to the
-   correctness of the final proof. *)
+(* NOTE: Assumption: order of merging segments satisfy associativity. *)
```

Comment:
Noted that the ranges use are starting from 0 here. But it seems internally we're counting from 1. 

May want to do it in a follow up PR to ensure consistency. Not a big deal, though.